src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -282,6 +282,27 @@ public class ReleaseService : IReleaseService
                 });
         }
 
+        public async Task<Either<ActionResult, List<MyReleaseViewModel>>> GetMyScheduledReleasesAsync()
+        {
+            return await _userService
+                .CheckCanAccessSystem()
+                .OnSuccess(_ =>
+                {
+                    return _userService
+                        .CheckCanViewAllReleases()
+                        .OnSuccess(() => _repository.GetAllReleasesForReleaseStatusesAsync(ReleaseStatus.Approved))
```

Comment:
Please can you explain why we attempt to get releases not related to the user before getting releases related to the user if that fails, because I didn't follow that? 😺

Reply:
Discussed in slack